### ISSUE TITLE
[Table] When `numbered` ensure the first header cell is not blank (CAL-AMP56)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#13544]
**monday.com sync:** #10821959707

### Violation: 
 Ensure data table headers cells are not blank 

### Recommended solution
In scenarios where we have an empty table cell in the header, for instance when the component is `numbered` we should provide context to assistive technologies of the numbered list row.

<img width="723" height="288" alt="Image" src="https://github.com/user-attachments/assets/b0837413-f18d-4b4b-831d-5d464f9aa95f" />

### WCAG Reference: 
- [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.html) 

### Severity: 
 Low 

### Media Type: 
 Data Tables 

--- 
### Areas for Remediation: 
- Title: CAL-AMP56 Ensure data table headers cells are not blank 
- Module: M7 Table and color picker 
- Finding ID: [CAL-AMP56](https://esri.hub.essentia11y.com/short-link/BJLlJq8ZgywrQLiw) 

**Issue**
There is an empty table header(present besides the "Checkbox" column). 

**User Impact**
Screen reader users will be unable to understand the purpose of their associated columns or rows.

**Code Reference** 
 ```
<th aria-colindex="1" class=" number-cell " role="columnheader" scope="col" tabindex="0"><!--?lit$792339209$--><!--?lit$792339209$--><!--?lit$792339209$--><!--?lit$792339209$--><span aria-live="off" class="assistive-text"><!--?lit$792339209$-->Row number</span></th>

```

**Steps to Reproduce**
1. Locate the table.
2. Inspect it with Chrome DevTools.
3. Notice that a ```<th>``` element does not contain content.
 
**Recommendation**
Ensure data table headers cells are not blank. The best way to resolve this issue is to add visual text to the table header. Alternatively, this text can be hidden with a .visually-hidden or .sr-only CSS class.

**Compliant Code Example**
```
<table>
 <caption>Grading scale</caption>
 <tr>
 <th scope="col">Letter Grade</th>
 <th scope="col">% Range</th>
 </tr>
 <tr>
 <td>A</td>
 <td>90 - 100%</td>
 </tr>
 <tr>
 <td>B</td>
 <td>80 - 89%</td>
 </tr>
 <tr>
 <td>C</td>
 <td>79 - 70%</td>
 </tr>
 <tr>
 <td>D</td>
 <td>69 - 60%</td>
 </tr>
 <tr>
 <td>F</td>
 <td>59 - 0%</td>
 </tr>
 </table>
```

 **Issue Image** 
 https://desktopqa/aic_lap/Calcite/Calcite2025/CAL-AMP56.png 
 

 --- 
### Additional Resources: 
- [LA Platform Report Link](https://esri.hub.essentia11y.com/short-link/BJLlJq8ZgywrQLiw) 
- [Esri Accessibility Resources](https://esriis.sharepoint.com/sites/Accessibility)
- Report Source: Product accessibility evaluation conducted on Calcite by Level Access and delivered August, 2025.
